scripts/table-s1-modality-summary.R

```diff
@@ -0,0 +1,109 @@
+  dplyr::filter(scpca_project_id %in% project_whitelist)
+
+# read in sample metadata and filter to included projects
+sample_metadata_df <- readr::read_tsv(sample_metadata_file) |> 
+  dplyr::filter(scpca_project_id %in% project_whitelist) |> 
+  dplyr::select(scpca_project_id, scpca_sample_id, diagnosis) |> 
+  # add diagnosis groupings
+  dplyr::left_join(diagnosis_groupings_df, by = c("diagnosis" = "submitted_diagnosis"))
+
+# Create table -----------------------------------------------------------------
+
+# create a list of all the modality values
+# we will use this to total the number of libraries for a project across all modalities 
+# these will also correspond to the column names in the final table that contain the total for each modality 
+modality_columns <- c("Single-cell", "Single-nucleus", "Bulk RNA", "Spatial transcriptomics", "With CITE-seq", "With cell hashing")
+
+demuxed_metadata_df <- library_metadata_df |> 
+  # make sure we have one row per library/ sample combination
+  # this specifically separates sample IDs for multiplex libraries for easier joining to sample metadata
+  dplyr::mutate(scpca_sample_id = stringr::str_split(scpca_sample_id, ";")) |> 
+  tidyr::unnest(scpca_sample_id)
+  
+# create a total count of the number of libraries per project per diagnosis group
+total_count <- demuxed_metadata_df |> 
+  dplyr::left_join(sample_metadata_df) |> 
+  dplyr::select(scpca_project_id, scpca_library_id, diagnosis_group) |> 
+  # remove any duplicates (e.g. those with CITE or with cell hash)
+  dplyr::distinct() |> 
+  dplyr::count(scpca_project_id, diagnosis_group, name = "Total number of libraries")
+
+modality_counts_df <- demuxed_metadata_df |> 
+  dplyr::mutate(
+    # make a new column that summarizes all modality info
+    # options are cell, nucleus, bulk, spatial, cite, or multiplexed
+    # the assigned values will eventually be the column names used in the output
+    modality = dplyr::case_when(
+      seq_unit == "bulk" ~ "Bulk RNA",
+      seq_unit == "spot" ~ "Spatial transcriptomics",
+      stringr::str_detect(technology,"CITE") ~ "With CITE-seq",
+      stringr::str_detect(technology,"cellhash") ~ "With cell hashing",
+      seq_unit == "cell" ~ "Single-cell",
+      seq_unit == "nucleus" ~ "Single-nucleus",
+    )
+  ) |> 
+  # join with sample metadata to add in diagnosis groupings
+  dplyr::left_join(sample_metadata_df) |>
+  dplyr::select(
+    scpca_project_id, scpca_library_id, modality, diagnosis, diagnosis_group
+  ) |> 
+  # for each project, group by diagnosis group, count the total number of libraries
+  # and then summarize the diagnoses included 
+  dplyr::group_by(scpca_project_id, diagnosis_group) |> 
+  dplyr::mutate(diagnosis = paste(unique(diagnosis), collapse = ";")) |>
+  # make sure that diagnosis now gets retained throughout creation of the table
+  dplyr::group_by(scpca_project_id, diagnosis_group, diagnosis) |> 
+  dplyr::count(scpca_project_id, modality) |> 
```

Comment:
You can just count here, which will also do (and then get rid of, which is good!) the grouping. The outcome looks the same to me in the end, unless I've missed something?

```suggestion
  dplyr::count(scpca_project_id, diagnosis_group, diagnosis, modality) |> 
```